src/framework/preferences/js/SyllabificationEnactor.js

```diff
@@ -218,29 +215,72 @@ var fluid_3_0_0 = fluid_3_0_0 || {};
         var hyphenator = fluid.getGlobalValue(globalPath);
 
         // If the pattern file has already been loaded, return the hyphenator.
+        // This could happen if the pattern file is statically linked to the page.
         if (hyphenator) {
             promise.resolve(hyphenator);
             return promise;
         }
 
         var src = fluid.stringTemplate(pattern, that.options.terms);
+
         var injectPromise = that.injectScript(src);
         injectPromise.then(function () {
             hyphenator = fluid.getGlobalValue(globalPath);
             promise.resolve(hyphenator);
-        }, function (jqXHR, textStatus, errorThrown) {
-            that.events.onError.fire(
-                "The pattern file \"" + src + "\" could not be loaded.",
-                jqXHR,
-                textStatus,
-                errorThrown
-            );
-            // If the pattern file could not be loaded resolve the promise with out a hyphenator (undefined).
+        }, function () {
+            var errorMessage = fluid.stringTemplate(that.options.strings.patternLoadError, {src: src});
+            // Pass along arguments fired for the rejection, but with our errorMessage as the first argument
+            var args = [errorMessage].concat(fluid.values(arguments));
+            that.events.onError.fire.apply(null, args);
+
+            //TODO: A promise rejection would be more appropriate. However, we need to know when all of the hyphenators
+            //      have attempted to load and apply syllabification. The current promise utility,
+            //      fluid.promise.sequence, will reject the whole sequence if a promise is rejected, and prevent us from
+            //      knowing if all of the hyphenators have been attempted. We should be able to improve this
+            //      implementation once https://issues.fluidproject.org/browse/FLUID-5938 has been resolved.
+            //
+            // If the pattern file could not be loaded, resolve the promise without a hyphenator (undefined).
             promise.resolve();
         });
+
         return promise;
     };
 
+    /**
+     * Information about a pattern, including the resolved language code and the file path to the pattern file.
+     *  {"lang": "resolvedLanguageCode", src: "pattern/file/path"}
+     *
+     * @typedef {Object} PatternInfo
```

Comment:
It's possible and desirable to document the members structurally using @property rather than in a comment - see examples at https://jsdoc.app/tags-typedef.html

Reply:
I modified this to use `@property`. I also fixed up a couple other typedefs that had a similar issue.